packages/compass-sidebar/src/components/multiple-connections/active-connections/active-connection-list.tsx

```diff
@@ -0,0 +1,108 @@
+  flexGrow: 0,
+  display: 'flex',
+  flexDirection: 'row',
+  alignContent: 'center',
+  justifyContent: 'space-between',
+});
+
+const activeConnectionListHeaderTitleStyles = css({
+  marginTop: 0,
+  marginBottom: 0,
+  textTransform: 'uppercase',
+  fontSize: '12px',
+});
+
+const activeConnectionCountStyles = css({
+  fontWeight: 'normal',
+});
+
+const activeConnectionsListStyles = css({
+  listStyle: 'none',
+  height: '100%',
+});
+
+export function ActiveConnectionList({
+  testUseActiveConnections,
+}: {
+  testUseActiveConnections?: () => ConnectionInfo[];
+}): React.ReactElement {
+  //eslint-disable-next-line react-hooks/rules-of-hooks
+  const activeConnections = testUseActiveConnections
+    ? testUseActiveConnections()
+    : useActiveConnections();
+  const connectionsCount = activeConnections.length;
+  const [collapsed, setCollapsed] = useState<string[]>([]);
+  const [namedConnections, setNamedConnections] = useState<
+    (ConnectionInfo & { title: string })[]
+  >([]);
+
+  const onConnectionToggle = (connectionId: string, isExpanded: boolean) => {
+    if (!isExpanded && !collapsed.includes(connectionId))
+      setCollapsed([...collapsed, connectionId]);
+    if (isExpanded && collapsed.includes(connectionId)) {
+      const index = collapsed.indexOf(connectionId);
+      setCollapsed([
+        ...collapsed.slice(0, index),
+        ...collapsed.slice(index + 1),
+      ]);
+    }
+  };
+
+  useEffect(() => {
+    // cleanup connections that are no longer active
+    // if the user connects again, the new connection should be expanded again
+    const newCollapsed = activeConnections
+      .filter(({ id }: ConnectionInfo) => collapsed.includes(id))
+      .map(({ id }: ConnectionInfo) => id);
+    setCollapsed(newCollapsed);
+
+    const newConnectionList = activeConnections
+      .map((connectionInfo) => ({
+        ...connectionInfo,
+        title: getConnectionTitle(connectionInfo),
+      }))
+      .sort((a, b) => a.title.localeCompare(b.title));
+    setNamedConnections(newConnectionList);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [activeConnections]);
+
+  return (
+    <div className={activeConnectionsContainerStyles}>
+      <header className={activeConnectionListHeaderStyles}>
+        <Subtitle className={activeConnectionListHeaderTitleStyles}>
+          Active connections{' '}
+          <span className={activeConnectionCountStyles}>
+            ({connectionsCount})
+          </span>
+        </Subtitle>
+      </header>
+      <ul className={activeConnectionsListStyles}>
+        {namedConnections.map((connection) => (
+          <ActiveConnection
+            key={connection.id}
+            connection={connection}
+            isExpanded={!collapsed.includes(connection.id)}
+            onToggle={(isExpanded: boolean) =>
+              onConnectionToggle(connection.id, isExpanded)
+            }
+          />
+        ))}
+      </ul>
```

Comment:
This is already better than the current saved connections list as your ActiveConnection component is following some of the accessibility guidelines, but this should be using our navigation tree component instead: you are basically just partially re-implementing some parts of it instead of making it work here. Is this temporary and we are planning to replace this with our navigation tree later or what's the plan here? Asking because once again we seem to add a lot of code that we'll just throw away later and have to rewrite.

Reply:
Yeah actually I planned to talk to you about this - originally I thought we could just implement this top level list and link the nested database lists, but I've realised this won't be that simple and we might need to redo the `compass-databases-navigation`, or extend it to include the new highest level. For this purpose I separated https://jira.mongodb.org/browse/COMPASS-7773.

Reply:
Sure, we can chat about this, feel free to set up a meeting or reach out in Slack. Some code that gives you an exact container size (like autosizer library does) is needed every time you are doing any sort of virtualization like this in a container of variable size, it's not in any way library dependent.

It's good to know that we have a follow up ticket, but I'm still confused that we are doing more work that we will be immediately re-doing.